src/projects/TodoList/index.js

```diff
@@ -37,46 +38,86 @@ class TodoList extends React.Component {
     }
   }
   changeCompleted(i) {
-    let taskListToBeAltered = this.state.todos
-    let alteredTask = taskListToBeAltered[i]
-    alteredTask.completed = !alteredTask.completed;
-    // taskListToBeAltered.splice(i, 1, alteredTask);
-    this.setState({ todos: taskListToBeAltered });
+    this.state.todos[i].completed = !this.state.todos[i].completed
+    // let taskListToBeAltered = this.state.todos
+    // let alteredTask = taskListToBeAltered[i]
+    // alteredTask.completed = !alteredTask.completed;
+    this.setState({ todos: this.state.todos });
+  }
+  showEditField(i) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.isEditing = !alteredTask.isEditing;
+    alteredTask.editingTask = alteredTask.task;
+    this.setState({ todos: this.state.todos });
+  }
+  confirmEditTask(i) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.task = alteredTask.editingTask;
+    alteredTask.isEditing = !alteredTask.isEditing;
+    this.setState({ todos: this.state.todos });
+  }
+  whilstEditing(i, value) {
+    let alteredTask = this.state.todos[i];
+    alteredTask.editingTask = value;
+    this.setState({ todos: this.state.todos });
   }
   render() {
     // Before the return you can do all logic you need
+    let areAnyTodosBeingEdited = this.state.todos.find(item => {
+      return item.isEditing;
+    })
     let todos = this.state.todos.map((task, i) => {
       return (
         <Todo
           key={i}
-          task={task}
+          todo={task.task}
+          completed={task.completed}
+          isEditing={task.isEditing}
+          editingTask={task.editingTask}
           removeItem={() => {
             this.removeItem(i);
           }}
-          completed={task.completed}
           changeCompleted={() => {
-            this.changeCompleted(i)
+            this.changeCompleted(i);
+          }}
+          editTask={() => {
+            this.showEditField(i);
+          }}
+          confirmEditTask={() => {
+            this.confirmEditTask(i);
+          }}
+          whilstEditing={(e) => {
+            this.whilstEditing(i, e.target.value);
           }}
         />
       )
     })
 
     return (
-      <div className='container'>
+      <div className="container mt-1">
         <h1>Hey I'm a todo list:</h1>
```

Comment:
If you have special characters like `'` or `"` in actual text in HTML you're supposed to use something called a HTML 'entity' to display these characters.

https://developer.mozilla.org/en-US/docs/Glossary/Entity

Some helpful references: 

`&quot;` is `"`
`&apos;` is `'`

Reply:
good advice! done!